.ci/no_old_refs.sh

```diff
@@ -5,27 +5,66 @@ set -e
 # Script requires GITHUB_TOKEN env variable
 MENTIONED_ISSUES=/tmp/mentioned_issues
 CLOSED_ISSUES=/tmp/failed_issues
+LINKED_ISSUES_FORMATTED=/tmp/linked_issues
+
+# Linked issues that are mentioned in the code
+LINKED_ISSUES_MENTIONED=/tmp/linked_issues_mentioned
 API_GITHUB_PREFIX="https://api.github.com/repos"
 GITHUB_HOST="https://github.com"
+CHECKSTYLE_ISSUE_PREFIX="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
+MAIN_REPO="checkstyle/checkstyle"
+DEFAULT_BRANCH="master"
+
+# These are modified when event is of type pull_request
+if [ ! -z "$PR_HEAD_REPO_NAME" ]; then
+  MAIN_REPO=$PR_HEAD_REPO_NAME
+  DEFAULT_BRANCH=$GITHUB_HEAD_REF
+fi
 
 # collect issues where full link is used
-grep -IPohr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
-  | sed -e 's/.*github.com\///' >> $MENTIONED_ISSUES
+grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+  | perl -pe 's/:(?!\d).*github.com\//:/' >> $MENTIONED_ISSUES
 
 # collect checkstyle issues where only hash sign is used
-grep -IPohr "[Tt]il[l]? #\d{1,5}" . \
-  | sed -e 's/.*#/checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+grep -IPonr "[Tt]il[l]? #\d{1,5}" . \
```

Comment:
https://www.gnu.org/software/grep/manual/grep.html#index-_002dn
```
-n 
--line-number
Prefix each line of output with the 1-based line number within its input file. (-n is specified by POSIX.)
```

Why we need line numbers ?

Reply:
https://www.gnu.org/software/grep/manual/grep.html#index-_002dh

```
-h
--no-filename
Suppress the prefixing of file names on output. This is the default when there is only one file (or only standard input) to search.
```

Reply:
https://github.com/checkstyle/checkstyle/runs/5686520746?check_suite_focus=true
Now CI reports a clickable link instead of manually telling where the closed issue is referenced.
https://github.com/Vyom-Yadav/checkstyle/blob/updateNoClosedRefWorkflow/src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java#L241

In the link mentioned above, I need the filename (so `-h` was removed) and I need line number (#L241) so `-n` was added.